src/storage/accessors/FileDataAccessor.ts

```diff
@@ -188,7 +194,13 @@ export class FileDataAccessor implements DataAccessor {
     metadata.remove(RDF.terms.type, LDP.terms.Container);
     metadata.remove(RDF.terms.type, LDP.terms.BasicContainer);
     metadata.removeAll(DC.terms.modified);
-    metadata.removeAll(CONTENT_TYPE_TERM);
+    // When writing metadata for a document, do not remove the contentType when dealing with an unsupported media type.
+    // (unsupported media type: when link.contentType does not match metadata.contentType)
+    // This allows restoring the appropriate content type on data read (see getFileMetadata).
+    const isUnsupportedMediaTypeDocument = !isContainerPath(link.filePath) && link.contentType !== metadata.contentType;
+    if (!isUnsupportedMediaTypeDocument) {
+      metadata.removeAll(CONTENT_TYPE_TERM);
+    }
```

Comment:
```suggestion
    // When writing metadata for a document, only remove the content-type when dealing with a supported media type.
    // A media type is supported if the FileIdentifierMapper can correctly store it.
    // This allows restoring the appropriate content-type on data read (see getFileMetadata).
    const isSupportedMediaTypeDocument = link.contentType === metadata.contentType;
    if (!isSupportedMediaTypeDocument) {
      metadata.removeAll(CONTENT_TYPE_TERM);
    }
```

Missed the `!isContainerPath(link.filePath)` last time, but this should also not be necessary as containers don't have a content-type anyway.

Also did a small rewrite so there is no double negation which helps in readability.